### PR TITLE
Facing the configuration file against a JSON schema.

### PR DESCRIPTION
**Pull request by jsanemet**
Adding a step after manual validation is done on the configuration file that faces it against a schema in order to catch any error that might have been missed by the other checks. Error messages produced by this last check will not be as good as the others are, but it is better than allowing a broken configuration to pass on.

Extracted YAML from the sample configurations on the documentation.

Added E2E tests that check that the sample configurations are accepted by Cibyl, meaning that they do match the schema.

### CODE REVIEW

File: cibyl/_data/schemas/config.json [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+  "$defs": {
+    "jenkins": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "driver": {
+          "type": "string",
+          "const": "jenkins"
+        },
+        "url": {
```
Comment (cescgina) [inline cibyl/_data/schemas/config.json:78]: `url` and `driver` are mandatory for jenkins

File: cibyl/_data/schemas/config.json [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    },
+    "jjb": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "driver": {
+          "type": "string",
+          "const": "jenkins_job_builder"
+        },
+        "repos": {
```
Comment (cescgina) [inline cibyl/_data/schemas/config.json:103]: `driver` and `repos` are mandatory for jjb

File: cibyl/_data/schemas/config.json [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+    },
+    "es": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "driver": {
+          "type": "string",
+          "const": "elasticsearch"
+        },
+        "url": {
```
Comment (cescgina) [inline cibyl/_data/schemas/config.json:129]: `driver` and `url` should be required in elasticsearch

File: cibyl/_data/schemas/config.json [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+    },
+    "zuul": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "driver": {
+          "type": "string",
+          "const": "zuul"
+        },
+        "url": {
```
Comment (cescgina) [inline cibyl/_data/schemas/config.json:145]: You'll know better than me, but `driver` and `url` should be required for zuul

File: cibyl/_data/schemas/config.json [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+        "enabled": {
+          "type": "boolean"
+        }
+      }
+    },
+    "zuul.d": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "driver": {
```
Comment (cescgina) [inline cibyl/_data/schemas/config.json:164]: for zuuld `driver` and `repos` should be required

File: cibyl/_data/schemas/config.json (hunk 1 above)
Comment (jsanemet) [reply cibyl/_data/schemas/config.json:78]: Done.

File: cibyl/_data/schemas/config.json (hunk 2 above)
Comment (jsanemet) [reply cibyl/_data/schemas/config.json:107]: Done.

File: cibyl/_data/schemas/config.json (hunk 3 above)
Comment (jsanemet) [reply cibyl/_data/schemas/config.json:137]: Done.

File: cibyl/_data/schemas/config.json (hunk 4 above)
Comment (jsanemet) [reply cibyl/_data/schemas/config.json:157]: Done.

File: cibyl/_data/schemas/config.json (hunk 5 above)
Comment (jsanemet) [reply cibyl/_data/schemas/config.json:180]: Done.

File: cibyl/_data/schemas/config.json [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+        "url": {
+          "type": "string"
+        },
+        "username": {
+          "type": "string"
+        },
+        "token": {
+          "type": "string"
+        },
+        "cert": {
```
Comment (cescgina) [inline cibyl/_data/schemas/config.json:87]: `cert` might also be a string with the path to a file containing certificates. Is it possible to define a Union type in the schema?

File: cibyl/_data/schemas/config.json [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+      "additionalProperties": false,
+      "patternProperties": {
+        "^.*$": {
+          "type": "object",
+          "additionalProperties": false,
+          "patternProperties": {
+            "^.*$": {
+              "type": "object",
+              "additionalProperties": false,
+              "properties": {
```
Comment (cescgina) [inline cibyl/_data/schemas/config.json:18]: Systems can also have a `jobs_scope` property, which is a string

File: cibyl/_data/schemas/config.json (hunk 6 above)
Comment (jsanemet) [reply cibyl/_data/schemas/config.json:87]: Done.

File: cibyl/_data/schemas/config.json (hunk 7 above)
Comment (jsanemet) [reply cibyl/_data/schemas/config.json:18]: Done.

**Review by cescgina — APPROVED** [review]
Great addition for UX

**Pull request merged**